src/caliper/Annotation.cpp

```diff
@@ -136,35 +137,62 @@ Loop::end()
 // --- Annotation implementation object
 
 struct Annotation::Impl {
-    std::atomic<Attribute*> m_attr;
-    std::string             m_name;
-    int                     m_opt;
-    std::atomic<int>        m_refcount;
+    std::atomic<Attribute*>  m_attr;
+    std::string              m_name;
+    std::vector<cali_id_t>   m_metadata_keys;
+    std::vector<cali_variant_t>     m_metadata_values;
+    int                      m_opt;
+    std::atomic<int>         m_refcount;
+    Impl(const std::string& name, MetadataListType metadata, int opt)
+        : m_attr(nullptr), 
+          m_name(name), 
+          m_opt(opt),
+          m_refcount(1)
+        {
+            Caliper c;
+            if(!c.attribute_exists(m_name)){
```

Comment:
Can technically do that with `c.get_attribute(m_name) != Attribute::invalid`, then we wouldn't need another function. But `attribute_exists()` might be nice to have.

Reply:
I'll leave the function